pkg/service/keystore/storage.go

```diff
@@ -104,27 +104,56 @@ const (
 	awsKMSScheme = "aws-kms"
 )
 
-func NewEncryption(db storage.ServiceStorage, tx storage.Tx, cfg config.KeyStoreServiceConfig) (Encrypter, Decrypter, error) {
+// EnsureServiceKeyExists makes sure that the service key that will be used for encryption exists. This function is
+// idempotent, so that multiple instances of ssi-service can call it on boot.
+func EnsureServiceKeyExists(config config.KeyStoreServiceConfig, provider storage.ServiceStorage) error {
+	if config.MasterKeyURI != "" {
+		return nil
+	}
+
+	watchKeys := []storage.WatchKey{{
+		Namespace: namespace,
+		Key:       skKey,
+	}}
+
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
 
-	if len(cfg.MasterKeyURI) != 0 {
-		return NewExternalEncrypter(ctx, cfg)
-	}
-
-	// First, generate a service key
-	serviceKey, serviceKeySalt, err := GenerateServiceKey(cfg.MasterKeyPassword)
+	_, err := provider.Execute(ctx, func(ctx context.Context, tx storage.Tx) (any, error) {
```

Comment:
in this function is `any` used? I only see nil returned

Reply:
It is not. The signature still needs it because the execute method expects one of its parameters to be a function that implements a specific signature.